tutorials/retail_api_v2_add_fulfillment_places_java.md

```diff
@@ -133,62 +128,33 @@ Clone the Git repository with all the code samples to learn the Retail features
    cd ~/java-retail/samples/snippets  
    ```
 
-1. Set the following fields to send the `AddFulfillmentPlacesRequest` request:
+1. Go to the `getAddFulfillmentRequest()` method in `AddFulfillmentPlaces` class and check which fields need to be set in order to send the `AddFulfillmentPlacesRequest`:
     - `product`—the product name whose inventory information will be updated
     - `type`—the fulfillment type. You can set one of the [supported values](https://cloud.google.com/retail/docs/reference/rpc/google.cloud.retail.v2#addfulfillmentplacesrequest).
-    - `place_ids[]`—the store IDs for each of the fulfillment types
-    - `add_time`—the time when the fulfillment updates are pushed. It is used to prevent out-of-order updates on the fulfillment information. If this isn't provided, the internal system time will be used.
-    - `allow_missing`—if set to true and the product is not found, the fulfillment information will be retained for up to 24 hours and processed after the product is created.
-
-1. Open <walkthrough-editor-select-regex filePath="cloudshell_open/java-retail/samples/snippets/src/main/java/product/AddFulfillmentPlaces.java" regex="Add fulfillment request">product/AddFulfillmentPlaces.java</walkthrough-editor-select-regex> file and check the `AddFulfillmentPlacesRequest` request.
```

Comment:
please dont remove the link - you can place this as a first step here

Reply:
fixed